### ISSUE TITLE
How to abort a running osmosis?

### ISSUE THREAD

**Issue by ninozhang** [rc0x03/node-osmosis#16]
For some reason, I want to abort a running osmosis.
How to do it?

**Comment by bencrenshaw0313** [comment]
var osmosis = require('osmosis');
var request = require('request');

for (var i = 0; i < 10; i++) {
    request('http://google.com', function(err, res, body) {
        osmosis
        .parse(body)
        .set('title', 'title')
        .data(console.log)
    });
} this should fix your problem

**Comment by eugenioclrc** [comment]
Try using a flag var;

```javascript
var _OSMOSIS_CRAWL = true;

osmosis.get(...)
.set(..)
.then(function(context, data, next) {
  if (!_OSMOSIS_CRAWL) {
    return;
  }
  next(context, data);
})
.follow('@href')
.set()
.data(function(data){ ...})

```

**Comment by bencrenshaw0313** [comment]
var _OSMOSIS_CRAWL = true;

osmosis.get(...)
.set(..)
.then(function(context, data, next) {
  if (!_OSMOSIS_CRAWL) {
    return;
  }
  next(context, data);
})
.follow('@href')
.set()
.data(function(data){ ...})

—

**Comment by rc0x03** [comment]
I'm adding this feature to the next release. Osmosis 0.0.7 will include `.pause`, `.resume`, and `.stop`.

Stop all Osmosis instances:
```javascript
osmosis.get('example.com')
...
.data(function(data) {
    if (data.date < today)
        osmosis.stop();
})
```

Stop specific Osmosis instances:
```javascript
var instance =
osmosis.get('example.com')
...
.data(function(data) {
    if (data.date < today)
        instance.stop();
})
```

**Comment by rc0x03** [comment]
Fixed in 0.0.7
